org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/handlers/WorkspaceEventHandlerTest.java

```diff
@@ -140,8 +140,6 @@ public void testDeleteProjectFolder() throws Exception {
 		assertEquals(0L, diags.get(5).getDiagnostics().size());
 		assertEndsWith(diags.get(6).getUri(), "/AppTest.java");
 		assertEquals(0L, diags.get(6).getDiagnostics().size());
-		assertEndsWith(diags.get(7).getUri(), "/multimodule3");
```

Comment:
Some explanation about why deleting test cases here:

From the previous code, it's very strange that the diagnostics of `multimodule3` and `multimodule3/pom.xml` are published twice. 

When I was figuring out where the second calls came from, I found that in `WorkspaceDiagnosticsHandler` it's triggered because of the resource delta of `multimodule3/module1`:

![image](https://user-images.githubusercontent.com/6193897/84336523-055dd380-abca-11ea-99b0-e9acd20a84ba.png)

That's strange because we are removing the folder `module2` but it affects `module1` and from the delta content shows above, the Language Server does not know the existence of `.classpath`, `.project` and `.settings` of `module1`.

So I went to check what's happening when importing the project, and found:
![image](https://user-images.githubusercontent.com/6193897/84336706-6d141e80-abca-11ea-988c-199307a07706.png)

You can see that the previous implementation will not update the projects (the list is empty) when importing the Maven projects. Thus the `refreshLocal()` is not called and language server does not know the existence of those files. We can also confirm this by adding the following codes before `new WorkspaceEventsHandler(projectsManager, javaClient, lifeCycleHandler).didChangeWatchedFiles(params);`

```java
IProject multimodule3 = ProjectUtils.getProject("multimodule3");
multimodule3.refreshLocal(IResource.DEPTH_INFINITE, monitor);
waitForBackgroundJobs();
```

Not sure if this is a bug or not, but IMO the current behavior makes more sense to me.